packages/yoroi-extension/app/stores/ada/AdaTransactionsStore.js

```diff
@@ -2,23 +2,58 @@
 
 import Store from '../base/Store';
 import type {
-  GetTransactionsFunc,
+  GetTransactionsFunc, GetTransactionsResponse,
   RefreshPendingTransactionsFunc,
+  BaseGetTransactionsRequest,
 } from '../../api/common';
 import type { ActionsMap } from '../../actions/index';
 import type { StoresMap } from '../index';
+import WalletTransaction from '../../domain/WalletTransaction';
+import CardanoShelleyTransaction from '../../domain/CardanoShelleyTransaction';
+import type { RemoteTokenInfo } from '../../api/ada/lib/state-fetch/types';
 
 export default class AdaTransactionsStore extends Store<StoresMap, ActionsMap> {
 
-  refreshTransactions: GetTransactionsFunc = (request) => {
+  refreshTransactions: GetTransactionsFunc = async (request: BaseGetTransactionsRequest) => {
     const stateFetcher = this.stores.substores.ada.stateFetchStore.fetcher;
 
-    return this.api.ada.refreshTransactions({
+    const txs: GetTransactionsResponse = await this.api.ada.refreshTransactions({
       ...request,
       getTransactionsHistoryForAddresses: stateFetcher.getTransactionsHistoryForAddresses,
       checkAddressesInUse: stateFetcher.checkAddressesInUse,
       getBestBlock: stateFetcher.getBestBlock,
     });
+
+    try {
+      const tokenIds = new Set<string>();
+      txs.transactions.forEach((tx: WalletTransaction) => {
+        tx.amount.values.forEach(t => tokenIds.add(t.identifier));
+        tx.addresses.from.flatMap(a => a.value.values).forEach(t => tokenIds.add(t.identifier));
+        tx.addresses.to.flatMap(a => a.value.values).forEach(t => tokenIds.add(t.identifier));
+        if (tx instanceof CardanoShelleyTransaction) {
+          tx.withdrawals.flatMap(w => w.value.values).forEach(t => tokenIds.add(t.identifier));
+        }
+      });
+      const missingMetaTokenIds = [...tokenIds]
+        .filter(tokenId => !localStorage.getItem(`token-metadata-${tokenId}`))
+        .map(id => id.split('.')[0]);
+      const tokenInfo = await stateFetcher.getTokenInfo({
+        network: request.publicDeriver.getParent().getNetworkInfo(),
+        tokenIds: missingMetaTokenIds,
+      });
+      // $FlowFixMe[incompatible-call]
+      Object.entries(tokenInfo).forEach(([tokenId, tokenMeta]: [string, RemoteTokenInfo]) => {
+        localStorage.setItem(`token-metadata-${tokenId}`, JSON.stringify({
```

Comment:
Presumably your key needs the network id as well to avoid collisions with the testnet.
In general it's usually better to use IndexedDB for this kind of stuff, but I understand the IndexedDB setup for the Yoroi Extension is quite complicated

Reply:
@SebastienGllmt , good point about the network ID, absolutely! The plan is to move all this into the indexed DB in the next version as we just need to get this working for now, and indexed DB needs a bit more time, yes.